qiskit/transpiler/transpile_circuit.py

```diff
@@ -27,18 +32,31 @@ def transpile_circuit(circuit, transpile_config):
 
     Returns:
         QuantumCircuit: transpiled circuit
-    """
 
+    Raises:
+        TranspilerError: if transpile_config is not valid or transpilation incurs error
+    """
     # if the pass manager is not already selected, choose an appropriate one.
     if transpile_config.pass_manager:
         pass_manager = transpile_config.pass_manager
 
+    elif transpile_config.optimization_level:
+        level = transpile_config.optimization_level
+        if level == 0:
+            pass_manager = level_0_pass_manager(transpile_config)
+        elif level == 1:
+            pass_manager = level_1_pass_manager(transpile_config)
+        elif level == 2:
+            pass_manager = level_2_pass_manager(transpile_config)
+        elif level == 3:
+            pass_manager = level_3_pass_manager(transpile_config)
+        else:
+            raise TranspilerError("optimization level can only be a non-negative integer")
```

Comment:
I think this error should say the range of allowable options, e.g. `optimization level can only be 0, 1, 2 or 3`,  as currently this wouldn't tell the user what was wrong if they set the optimization to `4`